Explorer/Assets/DCL/Communities/CommunityDataService/CommunityDataService.cs

```diff
@@ -47,10 +60,48 @@ public CommunityDataService(IChatHistory chatHistory,
             communitiesDataProvider.CommunityCreated += OnCommunityCreated;
             communitiesDataProvider.CommunityDeleted += OnCommunityDeleted;
             communitiesDataProvider.CommunityLeft += OnCommunityLeft;
+            communitiesDataProvider.CommunityUpdated += OnCommunityUpdated;
 
             SubscribeToCommunitiesBusEventsAsync().Forget();
         }
 
+        private void OnCommunityUpdated(string communityId)
+        {
+            // Fire-and-forget fetch; keep method sync because the provider callback is sync
+            RefreshCommunityAsync(communityId).Forget();
+        }
+
+        private async UniTaskVoid RefreshCommunityAsync(string communityId)
+        {
+            var ct = (communitiesServiceCts = communitiesServiceCts.SafeRestart()).Token;
+
+            var result = await communitiesDataProvider
+                .GetCommunityAsync(communityId, ct)
+                .SuppressToResultAsync(ReportCategory.COMMUNITIES);
+
+            if (ct.IsCancellationRequested) return;
+
+            if (!result.Success)
+            {
+                ReportHub.LogWarning(ReportCategory.COMMUNITIES, $"Failed to refresh community {communityId}: {result.ErrorMessage}");
+                return;
+            }
+
+            var data = result.Value.data;
+            var channelId = ChatChannel.NewCommunityChannelId(data.id);
+
+            // Update cache atomically
+            communities[channelId] = new GetUserCommunitiesData.CommunityData
+            {
+                id = data.id, thumbnails = data.thumbnails, name = data.name, description = data.description,
+                privacy = data.privacy, role = data.role, ownerAddress = data.ownerAddress, membersCount = data.membersCount
+            };
```

Comment:
what do you think about making a constructor for `CommunityData ` to hide all these assignments inside the class itself?